swift/ql/test/query-tests/Security/CWE-135/StringLengthConflation.swift

```diff
@@ -0,0 +1,125 @@
+
+// --- stubs ---
+
+func print(_ items: Any...) {}
+
+typealias unichar = UInt16
+
+class NSString
+{
+    init(string: String) { length = string.count }
+
+    func character(at: Int) -> unichar { return 0 }
+    func substring(from: Int) -> String { return "" }
+    func substring(to: Int) -> String { return "" }
+
+    var length: Int
```

Comment:
Fields like these are normally declared like:
```swift
var length: Int {
  get {
    // code that computes the length
  }
}
```
which makes it a read-only "computed" property (as opposed to a "stored property") since there is no corresponding `get` implementation. Obviously, we can't implement `length` as a computed property, but we can still model the fact that length cannot be modified from outside the class by writing:
```swift
private(set) var length: Int
```

Reply:
Done.